### PR TITLE
Summary clicks

### LINKED ISSUE #383: Disclosure - Double clicking the summary element selects it
**Describe the current behavior**

When you double click on the summary element (in filtering, for example, the name of the filter) it highlights it and looks aesthetically not great.

**Describe the expected behavior**

The open/close behaviour that happens is okay, it's what would be expected and is the web native behaviour.  The highlight, though, looks unpleasant and is confusing - we should remove it.

**Possible solution**

We could use `user-select: none` to stop it from highlighting.  We should see if all summary elements should receive this treatment by adding a global style on the `summary` selector itself.

### PR DESCRIPTION
**Pull request by ludoboludo**
**Why are these changes introduced?**

Fixes #383 

**What approach did you take?**

~Use `user-select: none;`~ Instead we're thinking of customizing the background color upon `::selection`. Check the issue's comment for more info.

**Other considerations**
 
**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=126316904470)
- [Editor](https://os2-demo.myshopify.com/admin/themes/126316904470/editor)

**Things to test**

Try and select the text across the theme: 
- on buttons, 
- on buttons with secondary style,
- text,
- links,
- variants,

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by ludoboludo** [comment]
@Guillaumegranger1 would you maybe have time to review the approach ? Or let me know if I should get in touch with another designer 🙂  
You can test using the links in the PR description.

**Comment by Guillaumegranger1** [comment]
I like this a lot. There only seems to be an issue when I chose the `Background 1` Color scheme, because the highlight doesn't repurpose the color that's assigned to the text. Here's a [screen recording ](https://screenshot.click/07-06-uy4du-8zygc.mp4)of that.

**Comment by ludoboludo** [comment]
> I like this a lot. There only seems to be an issue when I chose the `Background 1` Color scheme, because the highlight doesn't repurpose the color that's assigned to the text. Here's a [screen recording ](https://screenshot.click/07-06-uy4du-8zygc.mp4)of that.

Should be taken care of now. I was using the `link` color rather than text.

**Comment by svinkle** [comment]
- ✅ `#414141` / `#d0d0d0`: `6.61` (default text)
- ✅ `#ffffff` / `#5a5a5a`: `6.89` (white on black)
- 🛑 `#ffffff` / `#5e75bf`:`4.39` (blue product badge)

There may be other instances and I didn't test all default colors.

1. <kbd>Opt</kbd> + <kbd>a</kbd> to select everything
2. Grab color values using [macOS Digital Color Meter](https://support.apple.com/en-ca/guide/digital-color-meter/welcome/mac)
3. Test values with [contrast-ratio.com](https://contrast-ratio.com/)

**Comment by ludoboludo** [comment]
I took a look and it seems good to me with the default colors. When trying the sales badge, you get a couple different color values for the white text. If I take the whitest part then it seems good: `rgb(88,113,194)` and `rgb(254,254,255)` for a `4.57` contrast ratio 🤔 
Not sure if it's fair to base ourselves on that though.

File: assets/template-giftcard.css [hunk 1]
```diff
@@ -140,6 +148,10 @@ h2,
   background-color: transparent;
 }
 
+.button--secondary::selection {
+  background-color: rgba(var(--color-base-outline-button-labels), 0.3);
```
[the remark this answers is not in the document]
Comment (ludoboludo) [reply assets/template-giftcard.css:152]: `.button--secondary` in `base.css` is setting the variable `--color-button-text` to a new color.

And for `.button::selection` the background color is set to the variable `--color-button-text`. So it updates it automatically 👌

**Review by kmeleta — APPROVED** [review]
LGTM

**Comment by ludoboludo** [comment]
> I took a look and it seems good to me with the default colors. When trying the sales badge, you get a couple different color values for the white text. If I take the whitest part then it seems good: `rgb(88,113,194)` and `rgb(254,254,255)` for a `4.57` contrast ratio 🤔
> Not sure if it's fair to base ourselves on that though.

@svinkle let me know what you think. If you want to take another look maybe and give a 👍  or 👎

**Pull request merged**